client/Client.py

```diff
@@ -39,6 +43,42 @@ def join_queue(self):
             self.socket.shutdown(socket.SHUT_RDWR)
             self.socket.close()
 
+    def handle_threads(self):
+        try:
+            while True:
+                end_threads = 1
+                for connection_thread in self.connections_threads:
+                    if connection_thread.status.is_connected == -1:
+                        connection_thread.status.to_kill = 1
+                    if connection_thread.status.is_connected != -1:
+                        end_threads = 0
+                if end_threads:
+                    for connection_thread in self.connections_threads:
+                        connection_thread.status.to_kill = 1
+                    for connection_thread in self.connections_threads:
+                        connection_thread.thread.join()
+                    return
+        except KeyboardInterrupt:
+            for connection_thread in self.connections_threads:
+                connection_thread.status.to_kill = 1
+            for connection_thread in self.connections_threads:
+                connection_thread.thread.join()
```

Comment:
Grab it in one for loop

Reply:
Couldn't do that, since that was my first approach. I've seen that this way program enters an infinite loop, not sure why though. It might have been some other minor errors in code.